### DISCUSSION TITLE
Quantity Discount Per Customization Not Supported in PrestaShop's Native Specific Prices

### DISCUSSION THREAD

**Discussion by wzzly** [PrestaShop/PrestaShop#37314] [Ideas]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

In PrestaShop, the Specific Price functionality does not support quantity-based discounts per unique customization. When a product with customizations (e.g., personalized text) is added to the cart, quantity discounts are calculated based on the total quantity across all customizations, rather than for each unique customization separately. This prevents us from applying quantity discounts to individual customizations as intended.

**Expected Behavior:**
When adding products with unique customizations to the cart, each customization group should have discounts applied individually according to the quantity-specific rules set in Specific Prices.

**Steps to Reproduce:**

Create a product that allows customization (e.g., add a text field).
Set up Specific Prices with quantity discounts (e.g., 10% off for 12 items, 20% off for 48 items, etc.).
Add 12 units of a customized version (e.g., “Product A - Custom Text 1”) to the cart.
Add another unique customization of the same product (e.g., “Product A - Custom Text 2”) with only one unit.
Observe that the quantity discount for 12 units applies incorrectly to both customizations rather than individually.

### Expected behavior

When adding multiple quantities of a product with unique customizations to the cart, each customization group should independently trigger quantity discounts based on its own quantity count, as per the specific price rules.

For example:

If a product has a specific price rule for a 10% discount at 12 units, this discount should only apply if each unique customization reaches 12 units.
Adding 12 units of "Product A - Custom Text 1" should apply the 10% discount to that customization only.
Adding a separate customization (e.g., "Product A - Custom Text 2") with fewer than 12 units should not receive the 10% discount, unless it also reaches the threshold of 12 units independently.
This approach ensures discounts apply individually for each unique customization rather than aggregating quantities across all customizations of the same product.

### Steps to reproduce

**Create a Customizable Product:**
- Go to the PrestaShop back office and create a product (e.g., "Product A").
- Enable customization options for the product (e.g., add a text field for personalized messages).

**Set Up Quantity-Based Specific Prices:**
In the Specific Prices settings for "Product A," set quantity discount rules, such as:
- 12 units = 10% discount
- 48 units = 20% discount
- 96 units = 30% discount

**Add Customizations to the Cart:**
- Go to the storefront, add "Product A" with one unique customization (e.g., "Custom Text 1") and select a quantity of 12 units.
- Confirm that the 10% discount is applied correctly to this customization.

**Add Another Unique Customization:**
- Add a second unique customization of "Product A" (e.g., "Custom Text 2") with a different quantity, such as 1 unit.

**Observe the Discount Application:**
- Check the cart to see if the discount applies individually. Notice that the 10% discount intended for 12 units is incorrectly applied across both customizations rather than separately for each.

This demonstrates that the quantity discount does not apply independently per customization, impacting pricing accuracy for products with unique customizations.

### PrestaShop version(s) where the bug happened

1.7.8.10

### PHP version(s) where the bug happened

8.0

**Comment by Hlavtox** [comment]
Your suggested expected behavior may not be what every merchant wants to have.

When I sell a customizable product, let's say a wallet with custom name.
I setup a discount 20% for everybody that buys at least 3 wallets.
You want this discount not to work?

**Comment by wzzly** [comment]
Well it should have an option in the Specific prices section to choose if you want to group by attribute (for example colors) and customization_id.
